src/osd/osd_types.cc

```diff
@@ -340,6 +341,7 @@ void osd_stat_t::encode(bufferlist &bl) const
   ::encode(os_perf_stat, bl);
   ::encode(up_from, bl);
   ::encode(seq, bl);
+  ::encode(num_pgs, bl);
```

Comment:
The point of this field is as a second out of band safety check.  Not whether the current map maps to this osd, but whether there is still data stored on the osd that hasn't been cleaned up yet.  It's a pretty strict check but I'm erring on the side of paranoid here!